Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -257,37 +257,56 @@ public HdAccount GetUnusedAccount(Wallet wallet, string password)
             this.SaveToFile(wallet);
             return newAccount;
         }
-        
+
+
+        public string GetExtPubKey(WalletAccountReference accountReference)
+        {
+            Guard.NotNull(accountReference, nameof(accountReference));
+
+            Wallet wallet = this.GetWalletByName(accountReference.WalletName);
+
+            // get the account
+            HdAccount account = wallet.GetAccountByCoinType(accountReference.AccountName, this.coinType);
+
+            return account.ExtendedPubKey;
+        }
+
         /// <inheritdoc />
         public HdAddress GetUnusedAddress(WalletAccountReference accountReference)
         {
+            return GetUnusedAddresses(accountReference, 1).First();
+        }
+
+        /// <inheritdoc />
+        public IEnumerable<HdAddress> GetUnusedAddresses(WalletAccountReference accountReference, int count)
+        {
             Guard.NotNull(accountReference, nameof(accountReference));
+            Guard.Assert(count > 0);
 
             Wallet wallet = this.GetWalletByName(accountReference.WalletName);
 
             // get the account
             HdAccount account = wallet.GetAccountByCoinType(accountReference.AccountName, this.coinType);
 
-            // validate address creation
-            if (account.ExternalAddresses.Any())
+            if (account.ExternalAddresses == null)
```

Comment:
account.ExternalAddresses will never be null as it's initialized in the account ctor, so you can remove this check.